cargo-miri/src/util.rs

```
@@ -252,3 +252,20 @@ pub fn debug_cmd(prefix: &str, verbose: usize, cmd: &Command) {
     }
     eprintln!("{prefix} running command: {cmd:?}");
 }
+
+/// Checks if Cargo's build-std setting is enabled for one or more crates.
+pub fn is_build_std_set() -> bool {
+    let cmd = cargo()
+        // TODO: remove the -Z argument when this subcommand is stabilized;
+        // see <https://github.com/rust-lang/cargo/issues/9301>
+        .args(&["-Zunstable-options", "config", "get", "--format=json-value", "unstable.build-std"])
+        .output()
+        .expect("failed to get Cargo config");
+    // TODO: Cargo will exit with an error code if the `unstable.build-std` key is not set. Should
+    // we still return false if an unrelated error has occurred?
```

Ideally we'd distinguish these cases. Which status code does cargo return when the key is absent? The least we can do is raise an error on any unexpected status code.

---

Well, I think Cargo returns 101 for most errors (unless I am misunderstanding your question). This includes when the key is absent but also, e.g., when the key `a.b` cannot be resolved in the first place because `a` exists and is not a table. I'm sure there are many other error cases for `cargo config get` which return 101.

Relevant Cargo code: https://github.com/rust-lang/cargo/blob/f5418fcfebe7d01536b891c8b2ceb32c4eaf098c/src/cargo/util/errors.rs#L324C1-L341C2

---

Okay, that's unfortunate.

But still you can then treat only error 101 as "absent" and all other codes as actual errors.